### DISCUSSION TITLE
headless service, why?

### DISCUSSION THREAD

**Discussion by str11ngfello** [devspace-sh/devspace#1924] [Q&A]
I'm not sure if there's a discord server for the devspace community (or slack group?) so I thought a discussion might be fine. 

I understand devspace is creating a "-headless" service for each of my services. But, I don't really know what I'd use it for. If I move my team over to devspace, which is my goal, they will be like, why do all of our 20 services have 20 headless versions. Can someone tell me why it's doing this by default and perhaps a way to disable this functionality if it's not necessary?

Thank you! 

<img width="819" alt="image" src="https://user-images.githubusercontent.com/877000/156182202-8e4ed09b-7a92-4b70-a89e-9db9a321747e.png">

**Comment by LukasGentele** [comment]
@str11ngfello You are likely using a persistent volume in your deployment, right? So it's deployed as StatefulSet which requires a headless service as described here: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations

PS: Slack channel is over here: https://slack.loft.sh/

**Reply by str11ngfello** [reply]
Thanks for the doc ref and quick reply!
